First cut of code

LukeSP opened the pull request:
creating pull request so Nigel can comment

nigelcharman commented on src/test/java/org/concordion/specification/expectedToFailInfoExtensionTest.java:
```diff
@@ -0,0 +1,4 @@
+package org.concordion.specification;
```
This can be removed

nigelcharman commented on src/main/java/org/concordion/ext/expectedtofailinfo/ExpectedToFailInfoExtension.java:
```diff
@@ -0,0 +1,91 @@
+        }
+    }
+
+    public void setStyle(String style) {
```
Set return type to `ExpectedToFailInfoExtension` and `return this;` from the method

nigelcharman reviewed (commented):
Hey Luke, with Concordion 2.2.0 now supporting Ignored as a status, it would be great if this extension supported it too (and probably Unimplemented in case people want to add a reason to that too).

However, when an example is ignored, it does not publish Before and After example events. Since this extension works down from the root element, it could be changed to use @AfterSpecification instead, which will then be more performant too, since it is only invoked at the end of the spec. 

Just thinking about how this would work. Is there any way to have different notes and reasons on different examples within a single spec?

The name of the spec could also then change to something like concordion-status-info-extension or concordion-status-embellisher-extension.

nigelcharman commented on README.md:
```diff
@@ -1 +1,13 @@
\ No newline at end of file
+# concordion-expected-to-fail-info-extension
+
+This [Concordion](www.concordion.org) extension displays a Note and a Reason in the corresponding specification/markdown, when the annotation expectedToFail is used.
```
I'd remove "/markdown" since the output is the specification (always in HTML)

nigelcharman commented on src/main/java/org/concordion/ext/expectedtofailinfo/ExpectedToFailInfoExtension.java:
```diff
@@ -0,0 +1,112 @@
+ */
+public class ExpectedToFailInfoExtension implements ConcordionExtension, ExampleListener {
+
+    private String STYLE = "font-weight: normal; text-decoration: none; color: #bb5050;";
```
CAPS should only be used for static final. Please use lower case for all these fields.

nigelcharman commented:
I committed a few changes to gradle too. Hope they haven't broken it for you. I've got a couple more changes I'm using locally so we can publish the extension to maven local.

I just saw that the reason can come from the Markdown text too, so that answers my previous question. Would be great to have an example with some different statuses we could try this with.

Also, I noticed the specs and fixture classes start with a lower-case character. For Java standards these should be upper-case.

nigelcharman commented:
Oh, also it would be great if there was an index page for the specification suite that called the other specs.

LukeSP commented:
Hey Nigel,

Ive also encountered something a bit weird, which im not sure was
happeneing before, would need to go back and double check, but it looks
like if you apply @Extensions on a fixture class and add an @Extension to
overwrite an extension's value, like im doing in the
OverwritingDefaultProperties Example, it hits the extension twice, with the
unfortunate result of wiping out all of the scenario steps. Im unsure how
to revert to an older version of the build.gradle so havnt checked if this
was the case before.

This is easily resolved by removing the @Extensions annotation from the
class, just leaving the overwriting @Extension Object itself as part of the
fixture, this seems to call the listener as well. Im sure i would of
noticed this before when implementing so can only assume its a concordion
change.

Let me know how i can go to anolder build file and ill try again, ill also
update the fixture with a note for now.

Luke.





On Sun, Jul 29, 2018 at 1:24 PM, Luke Pearson <luke.pearson08@gmail.com>
wrote:

> Hi Nigel,
>
> I ran into a problem with building, specifically, gradle doesnt seem to be
> able to find the how to deal with the wrapper block. If i coipy in the
> build script in its entirety (this one; https://raw.githubusercontent.
> com/concordion/concordion-extension-build/master/extension-build.gradle)
> and comment out the wrapper block it works fine... Not sure what the cause
> is here or how to fix it.
>
> In any case I can build like this to make the fixes and additions above,
> untill a resolution is found.
>
> Luke.
>
> On Sun, Jul 22, 2018 at 4:48 PM, Nigel Charman <notifications@github.com>
> wrote:
>
>> Oh, also it would be great if there was an index page for the
>> specification suite that called the other specs.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/concordion/concordion-expected-to-fail-info-extension/pull/1#issuecomment-406840951>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AjOogiYKFehJIk--ycTynhjLBhMFbgpaks5uJAQmgaJpZM4TVR1x>
>> .
>>
>
>

LukeSP committed:
Note added for for issue where multiple listeners are created when both @Extension and @Extensions are in/on a class
updated some
updated some syntax
added in some placeholders for further enhancements around other concordion statuses

nigelcharman committed:
Changed to use SpecificationProcessingListener. Luke to fix :)

LukeSP committed:
ran into some issues with the ide around compiler version for java so added source and target compatability in the build.
made changes to support Ignored and Unimplemented status
- added tests for showing each
- added setters for all string variables for full customization of the messages on the spec.

LukeSP committed:
a new name!?

nigelcharman committed:
Modified sourceCompatibility and targetCompatibility to fix gradle issue.
Also modified from Java 8 to Java 7.

nigelcharman committed:
Modified case of fields and updated comments

LukeSP committed:
readme update
put in a check for the message size, should be h1-h6 only. Are there any others we might want to allow?

LukeSP committed:
touchup

LukeSP committed:
Merge branch 'master' into First_Cut_of_code

LukeSP committed:
Renamed the Project to Concordion Status Info Extension and so updated all uses in code and documentation to reflect the new name
Renamed some variables for clarity
Removed value for note variable as not needed, decided to leave it there though (titleTextPrefix) as a possible field for static information addition to the spec

nigelcharman commented on README.md:
```diff
@@ -1,12 +1,12 @@
 # concordion-status-info-extension
 
-This [Concordion](www.concordion.org) extension displays a Note and a Reason in the corresponding specification/markdown, when the annotation expectedToFail is used.
+This [Concordion](www.concordion.org) extension displays a Note and a Reason in the corresponding specification, when the annotations ignored, unimplemented or expectedToFail are used.
```
```suggestion
This [Concordion](https://concordion.org/) extension displays a Note and a Reason in the corresponding specification, when the annotations ignored, unimplemented or expectedToFail are used.
```
Currently shows up as a broken relative link.

nigelcharman commented on README.md:
```diff
@@ -1,12 +1,12 @@
 # concordion-status-info-extension
 
-This [Concordion](www.concordion.org) extension displays a Note and a Reason in the corresponding specification/markdown, when the annotation expectedToFail is used.
+This [Concordion](www.concordion.org) extension displays a Note and a Reason in the corresponding specification, when the annotations ignored, unimplemented or expectedToFail are used.
```
```suggestion
This [Concordion](www.concordion.org) extension displays a Note and a Reason in the output, when the `status`  of specific examples in a specification is [set as](https://concordion.org/instrumenting/java/markdown/#implementation-status) `Ignored`, `Unimplemented` or `ExpectedToFail`.
```
The word `annotation` makes me think this is for the `@ExpectedToFail` etc annotations in Java code. I assume it's not applicable to those, but is only when specific examples in the spec have the `c:status` attribute set?

nigelcharman replied:
Should we allow people to set this on the `@ExpectedToFail` annotation too?

LukeSP replied:
Maybe in the next release?...

LukeSP committed:
requested changes to readme done. fixed the broken concordion link and mentioned all the statuses
